jiant/tasks/qa.py

```diff
@@ -608,3 +607,102 @@ def preprocess_qasrl_datum(cls, datum):
                 for verb_idx, verb_entry in datum["verbEntries"].items()
             ],
         }
+
+
+@register_task("commonsenseqa", rel_path="CommonsenseQA/")
+@register_task("commonsenseqa-easy", rel_path="CommonsenseQA/", easy=True)
+class CommonsenseQATask(MultipleChoiceTask):
+    """ Task class for CommonsenseQA Task.  """
+
+    def __init__(self, path, max_seq_len, name, easy=False, **kw):
+        """ """
+        super().__init__(name, **kw)
+        self.path = path
+        self.max_seq_len = max_seq_len
+
+        self.easy = easy
+        self.train_data_text = None
+        self.val_data_text = None
+        self.test_data_text = None
+
+        self.scorer1 = CategoricalAccuracy()
+        self.scorers = [self.scorer1]
+        self.val_metric = "%s_accuracy" % name
+        self.val_metric_decreases = False
+        self.n_choices = 5
+        self.label2choice_idx = {"A": 0, "B": 1, "C": 2, "D": 3, "E": 4}
+        self.choice_idx2label = ["A", "B", "C", "D", "E"]
+
+    def load_data(self):
+        """ Process the dataset located at path.  """
+
+        def _load_split(data_file):
+            questions, choices, targs, id_str = [], [], [], []
+            data = [json.loads(l) for l in open(data_file, encoding="utf-8")]
+            for example in data:
+                this_question = tokenize_and_truncate(
```

Comment:
Please rename these "this" variables :S

Reply:
what would be a better name?

Reply:
just "question" instead of this_question, and same for the others.

Reply:
I add "this" because multichoice task will have multiple answers. 
If we use "answer" for answers in an instance, and "answers" for all the answers in the dataset, that's kind of misleading.
If we use "answers" for answers in an instance, then we need a different name for all the answers.

Reply:
And if we use names like "choicess", that's opening the doorway to endless typo-caused bugs.

Reply:
I'm pretty highly against using definite articles in variable names (since to a code reviewer, what does "the" or "this" exactly entail?

Reply:
You could also convert this_choices to something like multiple_choices, and then the rest of the variables you can just not use the "this_" in front of them.

Reply:
better now?